include/nighthawk/common/termination_predicate.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Nighthawk {
+
+class TerminationPredicate;
+
+using TerminationPredicatePtr = std::unique_ptr<TerminationPredicate>;
+
+/**
+ * Determines if and how execution is terminated.
+ */
+class TerminationPredicate {
+public:
+  enum class Status {
+    PROCEED,   // indicates a test should proceed
```

Comment:
"test" terminology used in these comments as well.

Reply:
Amended!